lib/block_view_dao_coin_limit_order.go

```diff
@@ -1229,6 +1389,12 @@ func (bav *UtxoView) IsValidDAOCoinLimitOrder(order *DAOCoinLimitOrderEntry, isC
 		return RuleErrorDAOCoinLimitOrderCannotBuyAndSellSameCoin
 	}
 
+	// Validate operation type.
+	if order.OperationType != DAOCoinLimitOrderOperationTypeASK &&
```

Comment:
This check seems off? an order can only have one operation type, no?

Reply:
I think it's right. If the operation type != ASK and the operation type != BID, then we error out: invalid operation type. It's a double negative. Equivalent to `if !(operation type == ASK || operation type == BID)`.

Reply:
Yep - I read it wrong. Good stuff